tdp/cli/commands/plan/dag.py

```diff
@@ -124,6 +130,14 @@ def dag(
     with Dao(database_dsn, commit_on_exit=True) as dao:
         planned_deployment = get_planned_deployment(dao.session)
         if planned_deployment:
-            deployment.id = planned_deployment.id
+            confirmed = click.confirm(
+                "A deployment plan already exists, do you want to override it?"
+            )
+            if confirmed:
+                dao.session.delete(planned_deployment)
+                click.echo("Previous planned deployment has been overritten")
+            else:
+                click.echo("No new deployment plan has been created")
+                return
         dao.session.merge(deployment)
     click.echo("Deployment plan successfully created.")
```

Comment:
I suggest this instead so that it better stick to the current implementation:

```suggestion
            if confirmed := click.confirm("A deployment plan already exists, do you want to override it?"):
                deployment.id = planned_deployment.id
            else:
                click.echo("No new deployment plan has been created.")
                return
    dao.session.merge(deployment)
    click.echo("Deployment plan successfully created.")
```

If that's good for you, please propagate it to the other command.

Reply:
implemented almost the same, just removed `confirmed := ` on line 133 and equivalent for other files.